lib/posix/pthread_mutex.c

```diff
@@ -9,6 +9,8 @@
 #include <wait_q.h>
 #include <posix/pthread.h>
 
+struct k_spinlock _z_pthread_spinlock = {};
```

Comment:
Is this empty initializer really needed? I don't see such pattern to be used elsewhere, and thus it's confusing. (And the only effect would likely be putting that variable into .data instead of .bss section, again, why?)